macros/CompatBeta.ejs

```diff
@@ -0,0 +1,603 @@
+      });
+    }
+
+    if (support.flag) {
+      // TODO handle array of flags (rename flag to flags)
+      // see https://github.com/mdn/browser-compat-data/issues/546
+      notes.push({
+        icon: writeIcon('disabled'),
+        text: writeFlagsNote(support, browserId)
+      });
+    }
+
+    if (notes.length > 0) {
+      for (let note of notes) {
+        output += '<dd>';
+        output += note.icon;
+        output += ' ' + note.text;
+        output += '</dd>';
+      }
+    } else {
+      output += '<dd></dd>';
+    }
+  }
+
+  output += '</dl></section>';
+
+  return output;
+}
+
+/*
+For a single row, write all the cells that contain support data.
+(That is, every cell in the row except the first, which contains
+an identifier for the row,  like "Basic support".
+
+*/
+function writeCompatCells(supportData) {
+  let output = '';
+
+  for (let browserNameKey of displayBrowers) {
+    let needsNotes = false;
+    let support = supportData[browserNameKey];
+    let supportInfo = '';
+    if (support) {
+      if (Array.isArray(support)) {
+        // Take first support data
+        supportInfo += getCellString(support[0].version_added,
+                                     support[0].version_removed,
+                                     support[0].partial_implementation);
+        needsNotes = true;
+      } else {
+        supportInfo += getCellString(support.version_added,
+                                     support.version_removed,
+                                     support.partial_implementation);
+        if (support.notes || support.prefix || support.flag || support.alternative_name) {
+          needsNotes = true;
+        }
+      }
+    } else { // browsers are optional in the data, display them as "?" in our table
+      supportInfo += getCellString(null);
+    }
+
+    output += `<td class="bc-supports-${getSupportClass(support)} bc-browser-${browserNameKey}`;
+
+    if (needsNotes) {
+      output += ' bc-has-history';
+    }
+
+    output += `">${supportInfo}`;
+
+    if (needsNotes) {
+      output += writeCellIcons(support);
+      output += writeNotes(support, browserNameKey);
+    }
+    output += '</td>';
+  }
+
+  return output;
+}
+
+function writeCompatFeatureRow() {
+  let output = '';
+  for (let row of features) {
+    output += '<tr>';
+    let feature = Object.keys(row).map((k) => row[k])[0];
+    output += `<th scope="row">${writeFeatureName(row, feature)}</th>`;
+    output += `${writeCompatCells(feature.support)}</tr>`;
+    output += '</tr>';
+  }
+  return output;
+}
+
+function writeCompatBody() {
+  let output = '<tbody>';
+  output += writeCompatFeatureRow()
+  output += '</tbody>';
+  return output;
+}
+
+function writeCompatPlatformsRow() {
+  let output = '<tr class="bc-platforms">';
+  output += '<td></td>';
+
+  let platforms = ['desktop', 'mobile'];
+  if (category === 'javascript') {
+    platforms.push('server');
+  }
+  if (category === 'webextensions') {
```

Comment:
Can this switching on table type stuff be moved up to line 60ish where the other switching is done?

Reply:
Yes, good call. Changed that.